modules/nw-ingress-integrating-route-secret-certificate.adoc

```diff
@@ -0,0 +1,71 @@
+[id="nw-ingress-integrating-route-secret-certificate_{context}"]
+= Secure route with external certificates in TLS secrets
+
+:FeatureName: Secure route with external certificates in TLS secrets
+include::snippets/technology-preview.adoc[]
+
+The following workflow describes how to integrate third-party certificate management solutions, using the certificate reference field outlined under API Extensions.
+
+A new `.spec.tls.externalCertificate` field has been added to the route's `.spec.tls` configuration. This field can be used to specify the name of a secret containing the certificate data.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the cluster-admin role.
+
+* You must have generated the serving certificate using third-party systems.
+
+* Make sure the `secret` is categorized as `kubernetes.io/tls`, and include both `tls.key` and `tls.crt` keys within the data or `stringData` field of the secret configuration.
+
+.Procedure
+
+* Create a role in the same namespace as the secret containing the certificate generated.
++
+[source,terminal]
+----
+$ oc create role secret-reader --verb=get,list,watch --resource=secrets --resource-name=<secret-name>
+--namespace=<current-namespace>
+----
+
+* Create a rolebinding in the same namespace as the secret and bind the router service account to the newly created role.
++
+[source,terminal]
+----
+$ oc create rolebinding secret-reader-binding --role=secret-reader --serviceaccount=openshift-ingress:router --namespace=<current-namespace>
+----
+
+* To enable access to a user workload, generate a new Route referencing the generated secret in the .`spec.tls.externalCertificate` field from the previous step.
++
+.Example output
+[source,terminal]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: myedge
+  namespace: test
+spec:
+  host: myedge-test.apps.example.com
+  tls:
+    externalCertificate:
+      name: <secret-name>
+    termination: edge
+----
+
+If the referenced secret exists and has successfully generated a certificate/key pair, the router will serve this certificate, provided all prerequisites are fulfilled.
+
+[NOTE]
+====
+If neither `.spec.tls.externalCertificate` nor both `.spec.tls.certificate` and `.spec.tls.key` are provided, the router serves the default generated certificates.
+
+Users are unable to provide both `.spec.tls.certificate`/`.spec.tls.key` and `.spec.tls.externalCertificate`. 
+====
+
+.Troubleshooting
+
+If the router encounters an invalid route, it's considered an edge case. This could happen due to changes in role-based access control (RBAC) or if the referenced secret is deleted after API server validation but before the router processes the route, for example, if the router pod isn't running.
+
+To resolve this issue, it's essential to replicate the RBAC validations present on the API server within the router's functionality, particularly as part of the `ExtendedValidateRoute()` process. 
```

Comment:
I think it's not required, being implementation details.

Reply:
The only troubleshooting (given all the above steps were followed correctly), which I think is worth adding, is to check the OpenShift router pod logs for any errors.